assets/js/filter.js

```diff
@@ -0,0 +1,196 @@
+$(function() {
+    "use strict";
+
+    $('.filtered-table').aplusTableFilter();
+});
+
+/**
+ * Table row filter.
+ */
+(function($, window, document, undefined) {
+    "use strict";
+
+    var pluginName = "aplusTableFilter";
+    var defaults = {};
+    var translationsReady = false;
+
+    function AplusTableFilter(element, options) {
+        this.element = $(element);
+        this.filters = null;
+        this.timeout = null;
+        if (this.element.prop("tagName") == "TABLE") {
+            this.settings = $.extend({}, defaults, options);
+            var self = this;
+
+            // translationsReady allows us to use the plugin multiple times on same page
+            if (translationsReady) {
+                self.init();
+            }
+            $(document).on("aplus:translation-ready", function() {
+                translationsReady = true;
+                self.init();
+            });
+        }
+    }
+
+    $.extend(AplusTableFilter.prototype, {
+
+        init: function() {
+            var columnCount = 0;
+            this.element.find('thead').find('tr').each(function() {
+                var count = $(this).find('th').size();
+                columnCount = count > columnCount ? count : columnCount;
+            });
+
+            var self = this;
+            var filterDelay = function(event) {
+                var input = $(this);
+                clearTimeout(self.timeout);
+                self.timeout = setTimeout(function() {
+                    self.filterColumn(input);
+                }, 500);
+            };
+
+            this.filters = [];
+            var filterRow = $('<tr class="tableexport-ignore"></tr>');
+            for (var i = 0; i < columnCount; i++) {
+                this.filters.push('');
+                var filterInput = $('<input type="text" data-column="' + i + '">')
+                    .on('keyup', filterDelay).on('change', filterDelay);
+
+
+                var filterCell = $(
+                    '<td data-toggle="tooltip" title="'
+                    + _(
+                        "Comparison operators >, <, >=, <=, !=, == are available — e.g. >=200. "
+                        + "Use dot as decimal seperator — e.g. 2.1. "
+                        + "Regex is also available — e.g. /^\\d+$/g"
+                        )
+                    + '"></td>'
+                );
+                filterCell.append(filterInput);
+                filterRow.append(filterCell);
+            }
+            this.element.find('thead').append(filterRow);
+        },
+
+
+        filterGetType: function(query) {
+            if (!query) {
+                // Empty query, no filtering
+                return [undefined, ""];
+            }
+
+            var querySliceOne = query.slice(1);
+            var querySliceTwo = query.slice(2);
+            querySliceOne = isNaN(querySliceOne) ? querySliceOne : parseFloat(querySliceOne);
```

Comment:
Variables like "one" and "two" are not explaining what they are. You sohuld name them a bit better.